src/Cms/File.php

```diff
@@ -157,14 +158,26 @@ public function blueprint(): FileBlueprint
 	 */
 	public function blueprints(string $inSection = null): array
 	{
+		// get cached results for the current file model
+		// (except when collecting for a specific section)
 		if ($inSection === null && $this->blueprints !== null) {
 			return $this->blueprints; // @codeCoverageIgnore
 		}
 
+		$parent = $this->parent();
+
+		// get cached results from other models that share the same parent
+		// and thus the same available blueprints
+		// (except when collecting for a specific section)
+		static $cache = new WeakMap();
```

Comment:
This is what I meant with "not very Kirby like", I did it this way to keep the example implementation code to the body of a single function, but I think it could be more useful to cache the "accepted file blueprints" in the parent model, rather than statically between the children. 
This could probably be done by adding something like a dedicated "acceptedFileBlueprints" method to the `HasFiles` trait, and cache the result there.

Reply:
@lukasbestle maybe also good to get your perspective on this: should we move all the logics to a `HasFiles::acceptedFileBlueprints()` and `File::blueprints()` would just call `$this->parent()->acceptedFileBlueprints()`?

Reply:
I think it might actually be harder to do it on the parent model: these lines https://github.com/getkirby/kirby/pull/6112/files#diff-3b700032469a1feb9c9a9eab21e85c3b799a16b107327bc5173c1e375e962f06R179-R180 won't be able to do in a method of the parent... so we would have to run similar code twice.

Reply:
Oh, I see. 
It was just that `static` keyword inside of the function that smelled a bit for me; not sure why... Probably because it means only the `File::blueprints` function benefits from the cache.

I've gone a bit deeper, inside of the `blueprints()` function the problematic path is actually:
 - [`$blueprint->sections()`](https://github.com/getkirby/kirby/blob/4.0.3/src/Cms/File.php#L208C12-L208C34) 
 - [`$this->section($section['name'])`](https://github.com/getkirby/kirby/blob/main/src/Cms/Blueprint.php#L716C21-L716C53)
 - [`new Section($props['type'], $props)`](https://github.com/getkirby/kirby/blob/main/src/Cms/Blueprint.php#L706C10-L706C45)
 - [`parent::__construct($type, $attrs);`](https://github.com/getkirby/kirby/blob/main/src/Cms/Section.php#L46C3-L46C38)
 - [`$this->applyComputed($options['computed']);`](https://github.com/getkirby/kirby/blob/main/src/Toolkit/Component.php#L91C4-L91C46)
 - In there it's the [`data` computed prop of the `files` section](https://github.com/getkirby/kirby/blob/main/config/sections/files.php#L102-L138) 
 - In there it's split between `$panel->image(...)` (60%), `$file->mime(...)` (25%) and `$file->toSafeString(...)` (5%). The remaining 10% is more or less equally split between the rest. 

I guess the main problem is that for every file in the page, the `applyComputed` method is called for every file, twice: that's `O(2n²)`. So for 5 files the `data` computed prop of the files section is called 50 times (2*5²), for 100 files it's called 20'000 times. 

Options I see:
1. just go with the (perhaps smelly) local static WeakMap
2. split `File::blueprints` into a new `HasFiles::acceptedFileTemplates()` function call (which caches the results in the parent), then add the current file template and continue by starting from the line [`$templates = array_unique(array_filter($templates));`](https://github.com/getkirby/kirby/blob/4.0.3/src/Cms/File.php#L228).
3. have a cache for the `Section` **instances** in [`Blueprint::section(...)`](https://github.com/getkirby/kirby/blob/4.0.3/src/Cms/Blueprint.php#L693-L707), currently only the props are cached.

Reply:
3. -> tried something, haven't encountered side regressions so far: https://github.com/getkirby/kirby/pull/6175

Reply:
@rasteiner I tried no. 2 in this PR and also no 3 https://github.com/getkirby/kirby/pull/6175

Reply:
Haven't reviewed the full code of this PR so far, but at the first glance I very much like splitting up the method into multiple smaller ones. The previous/current implementation is *a lot* of logic in a single method.

Reply:
@lukasbestle That's already done